cryptomite/raz.py

```diff
@@ -0,0 +1,466 @@
+            ls = [i for i in range(best_l - max_minus, best_l + max_plus + 1)]
+            for current_l in ls:
+                p_half_max = int((2**(current_l - log2(m)))//2)
+                p_values = [2 * phalf + 2 for phalf in range(p_half_max)]
+                for current_p in p_values:
+                    eps = log2_error_raz(n_1,
+                                         k_1,
+                                         k_2,
+                                         m,
+                                         current_l,
+                                         current_p)
+                    if eps < min_log2_error:
+                        min_log2_error = eps
+                        best_l, best_p = current_l, current_p
+    if verbose:
+        print(f'log2 of minimum error found: {min_log2_error}')
+        print(f'Corresponding l value (p prime = 2^l): {best_l}')
+        print(f'Corresponding p value: {best_p}')
+    return min_log2_error
+
+
+def calc_raz_out(n_1: int,
+                 k_1: float,
+                 n_2: int,
+                 k_2: float,
+                 log2_error_tol: float,
+                 m_init=1,
+                 max_tests_basic=1,
+                 max_tests_detailed=1000,
+                 detailed_opt=False,
+                 verbose=False):
+    """
+    Compute the maximum output length for the efficient
+    weak version of Raz's extractor presented in
+    [Fore2025]_ that satisfies a given error tolerance.
+
+    Parameters
+    ----------
+    n_1 : int
+        The length of the first input (in bits).
+    k_1 : float
+        The min-entropy of the first input.
+    n_2 : int
+        The length of the second input (in bits).
+    k_2 : float
+        The min-entropy of the second input.
+    log2_error_tol : float
+        The logarithm base 2 of the acceptable extractor error.
+        Must be negative, as the extractor error is 2^log2_error.
+    m_init : int
+        The initial value for the output length (default: 1).
+    max_tests_basic : int
+        The maximum number of interations for the
+        basic parameter optimisation method, i.e. when
+        detailed_opt is set to False (default: 0).
+    max_tests_detailed : int
+        The maximum number of interations for the
+        intense parameter optimisation method, i.e.
+        when detailed_opt is set to True (default: 1000).
+    detailed_opt : bool
+        Flag to indicate the intensity of the
+        optimisation performed (default: False).
+    verbose : bool
+        If True, prints all parameters found (default: False).
+
+    Returns
+    -------
+    int :
+        The maximum output length that satisfies the
+        error tolerance.
+    """
+    max_m = 0
+    initial_tests = 100
+    ms = np.linspace(m_init, floor(k_2),
+                     min(floor(k_2) - m_init,
+                         initial_tests), dtype=int)
+    i = 0
+    STOP = 0
+    while STOP == 0:
```

Comment:
I think you can remove the `STOP` variable and instead use `while True:` then `break` when you want to terminate the loop.